src/core/lib/iomgr/ev_epollsig_linux.c

```diff
@@ -732,7 +732,7 @@ static void workqueue_maybe_wakeup(polling_island *pi) {
      it right now. Note that since we do an anticipatory mpscq_pop every poll
      loop, it's ok if we miss the wakeup here, as we'll get the work item when
      the next poller enters anyway. */
-  if (current_pollers > min_current_pollers_for_wakeup) {
+  if (current_pollers >= min_current_pollers_for_wakeup) {
```

Comment:
This should be the only real change (the rest is in bumping up logging to make these bugs more transparent in the future).